streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -763,10 +762,20 @@ private void runLoop() {
         consumer.subscribe(builder.sourceTopicPattern(), rebalanceListener);
 
         while (isRunning()) {
-            recordsProcessedBeforeCommit = runOnce(recordsProcessedBeforeCommit);
+            try {
+                recordsProcessedBeforeCommit = runOnce(recordsProcessedBeforeCommit);
+            } catch (final TaskMigratedException ignoreAndRejoinGroup) {
+                log.warn("Detected a task that got migrated to another thread. " +
+                    "This implies that this thread missed a rebalance and dropped out of the consumer group. " +
+                    "Trying to rejoin the consumer group now.");
```

Comment:
Should we close the zombie task before moving on to the next `runOnce` loop, cause otherwise we are doomed to get the TaskMigrationException again when trying to suspend the task in rebalance callback right? Or do you think it is better to stay as is to be more restrict for corner cases (like the fenced tasks suddenly become not fenced any more)?

Reply:
Yes, we will get the exception again, and this will triggering closing the zombies in `AssignedTasks#suspendTasks` automatically (ie, in `onPartitionsRevoked` callback). Not sure if it's worth closing the zombie here. I think for now it's fine -- we might revisit this after 1.0 release. To me, it's just important for 1.0 that the thread does not die.

Reply:
After thinking about this once more, it seems reasonable to close zombie task immediately. I added corresponding code when we throw the exception. I cannot close zombies here, because we might loop through all task performing some actions, and not get all exceptions here, but only the first one of a loop.